clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -603,6 +606,17 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                             TESTANY(OFFLINE_FILE_TYPE_KERNEL_SYSCALL_TRACE_TEMPLATES,
                                     shard->file_type_),
                         "Mismatching syscall num in trace end and syscall marker");
+#ifdef UNIX
+        if (shard->between_kernel_syscall_trace_markers_) {
+            // Protected by above if-condition to avoid noise from another invariant
+            // error due to a missing start marker.
+            report_if_false(shard,
+                            shard->signal_stack_depth_at_syscall_trace_start_ ==
+                                static_cast<int>(shard->signal_stack_.size()),
+                            "Syscall trace has extra kernel_event marker");
```

Comment:
Worth a comment announcing/explaining that we expect perfect pairing of kernel event+xfer in syscall templates, since that is not obvious that the kernel would guarantee that.

Reply:
I did get a perfect pairing of interrupts and interrupt-returns in the syscall injection templates I collected on QEMU.

I'm not so sure about the PT traces, but we don't make an effort to add kernel_event and kernel_xfer markers at all in PT traces today.

Added a comment.